app/models/notification.rb

```diff
@@ -59,12 +74,18 @@ def self.purge_old!
     SQL
   end
 
+  def self.consistency_exempt_sql
+    return "" if @consistency_exempt_types.empty?
+
+    "AND notification_type NOT IN (#{@consistency_exempt_types.join(", ")})"
+  end
+
   def self.ensure_consistency!
     DB.exec(<<~SQL)
       DELETE
         FROM notifications n
        WHERE high_priority
-         AND notification_type NOT IN (#{types[:chat_mention].to_i}, #{types[:chat_message].to_i})
+         #{consistency_exempt_sql}
```

Comment:
I'm thinking about why we need this exemption in the first place. Chat notifications are not attached to topics in anywhere. Perhaps this SQL query should just by default be ignoring notifications which has `Notification#topic_id` set to null?

Reply:
Yeah I think this is a good idea.